blitz-js/blitz
# Add `blitz install graphql-apollo-server` recipe to implement a GraphQL API endpoint

### What are the changes and their implications?

- New `blitz install graphql` recipe which sets up a GraphQL endpoint + Playground at `http://localhost:3000/api/graphql`
- New addBabelPlugin() transform to make it easier to install babel plugins in recipes:

```js
/* Supports simple string style and array with options */

// plugins: ["@emotion"]
return addBabelPlugin(program, "@emotion")

// plugins: [["@babel/plugin-proposal-decorators", {legacy: true}]]
return addBabelPlugin(program, ["@babel/plugin-proposal-decorators", {legacy: true}])
```

**Testing Notes:**

1. Create a new blitz app `blitz new graphql-test`
2. Install this recipe `blitz install graphql`
3. Start the dev server `yarn dev`
4. Visit http://localhost:3000/ and register a new User account
5. Visit http://localhost:3000/api/graphql and perform the following query:

```gql
query {
  allUsers {
    id
    email
    role
  }
}
```

  Your user information should appear!

### Checklist

- [x] Changes covered by tests (tests added if needed)
- [ ] PR submitted to [blitzjs.com](https://github.com/blitz-js/blitzjs.com) for any user facing changes
